device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubListener.java

```diff
@@ -60,8 +60,10 @@
      * @param connectionId the Id of the connection, used to identify which of possible many reconnection attempts
      *                     this event belongs to.
      * @param deviceId the Id of the device that the session belongs to
+     * @param shouldReconnect false if this session was lost because the user opted to unregister the client from
+     * the multiplexed connection.
      */
-    void onMultiplexedDeviceSessionLost(TransportException e, String connectionId, String deviceId);
+    void onMultiplexedDeviceSessionLost(TransportException e, String connectionId, String deviceId, boolean shouldReconnect);
```

Comment:
Part of the bug I'm solving here is tied to ambiguity around when a device session closes. This new flag makes it clear that the session was either lost because it was closed by the user, or because it was closed as a part of a broader reconnection